app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from "react";
+
+import { Formik, Form } from "formik";
+import { Check } from "neetoicons";
+import { Button, Pane, Toastr } from "neetoui";
+import { Input, Select } from "neetoui/formik";
+
+import {
+  CONTACTS_FORM_VALIDATION_SCHEMA,
+  ROLES_DUMMY_DATA,
+} from "../constants";
+
+const ContactForm = ({ onClose, contact, isEdit }) => {
+  const [submitted, setSubmitted] = useState(false);
+
+  const handleSubmit = () => {
+    Toastr.success("Contact added successfully");
+    onClose();
+  };
+
+  return (
+    <Formik
+      initialValues={contact}
+      validateOnBlur={submitted}
+      validateOnChange={submitted}
+      validationSchema={CONTACTS_FORM_VALIDATION_SCHEMA}
+      onSubmit={handleSubmit}
+    >
+      {({ isSubmitting }) => (
+        <Form className="w-full">
+          <Pane.Body className="space-y-6">
+            <div className="flex w-full justify-between space-x-4">
+              <Input
+                required
+                className="flex-grow-1"
+                label="First Name"
+                name="firstName"
+                placeholder="Enter first name"
+              />
+              <Input
+                required
+                className="flex-grow-1"
+                label="Last Name"
+                name="lastName"
+                placeholder="Enter last name"
+              />
+            </div>
+            <Input
+              required
+              className="w-full flex-grow-0"
+              label="Email"
+              name="email"
+              placeholder="Enter your email address"
+            />
+            <Select
+              isSearchable
+              required
+              className="w-full flex-grow-0"
+              label="Role"
+              name="role"
+              placeholder="Select Role"
+              options={ROLES_DUMMY_DATA.map(({ id, name }) => ({
+                label: name,
+                value: id,
+              }))}
+            />
+          </Pane.Body>
+          <Pane.Footer>
+            <Button
+              className="mr-3"
+              disabled={isSubmitting}
+              icon={Check}
+              label={isEdit ? "Update" : "Save Changes"}
+              loading={isSubmitting}
+              size="large"
+              style="primary"
+              type="submit"
+              onClick={() => setSubmitted(true)}
+            />
+            <Button
+              label="Cancel"
+              size="large"
+              style="text"
+              onClick={onClose}
+            />
```

Comment:
Go through the Slack thread.

Reply:
Did you mean to set the type prop as "reset"?
Shouldn't the cancel button be used to hide the pane? in the examples in the neetoui docs, the cancel button hides the pane.
Or should I rename the button as "Reset"

Reply:
@anandhakrishnanaji The cancel hides the pane and in turn resets the data. From formik perspective this is a button that can reset the data. Thus it's better to be explicit here and set the type as `reset`.